code/datums/components/remote_materials.dm

```diff
@@ -76,14 +77,34 @@ handles linking back and forth.
 	if (!silo && mat_container)
 		mat_container.max_amount = size
 
-// called if disconnected by ore silo UI or destruction
+// called if disconnected by ore silo UI, or destruction
 /datum/component/remote_materials/proc/disconnect_from(obj/machinery/ore_silo/old_silo)
 	if (!old_silo || silo != old_silo)
 		return
 	silo = null
 	mat_container = null
 	if (allow_standalone)
 		_MakeLocal()
+	return TRUE
+
+/datum/component/remote_materials/proc/check_z_disconnect()
+	SIGNAL_HANDLER
+	if(!silo) //No silo?
+		return
+	var/atom/P = parent
+	if((!is_station_level(P.z) && !is_station_level(silo)) || (P.get_virtual_z_level() != silo.get_virtual_z_level()))
```

Comment:
This will break if you cross station z-levels.

You get disconnected if:
 - The silo is not on the station, and you are not moving on the station
or
 - You change virtual Zs.

Imo it should be if you change virtual Zs and (you or the silo are not on the station level.)

Reply:
```suggestion
	if((!is_station_level(P.z) && !is_station_level(silo)) && (P.get_virtual_z_level() != silo.get_virtual_z_level()))
```